lib/azure/core/http/http_request.rb

```diff
@@ -149,7 +154,7 @@ def call
 
           response = HttpResponse.new(res)
           response.uri = uri
-          raise response.error unless response.success?
+          raise response.error if !response.success? && !@has_retry_filter
```

Comment:
The code in the `call(req, _next)` method in line 37 of retry_policy.rb could NOT deal it properly.
Because `response` can never be passed to the `should_retry` method when there is an error in the response, as it will raise exception before calling `should_retry` (as this line of code does).

With `!@has_retry_filter`, it will not raise exception in this method, it will set `retry_data[:error]` with the error code and raise the error in line 46 of retry_policy.rb when reties fail.